src/Shared/SqlProcessor.cs

```diff
@@ -0,0 +1,234 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text;
+
+namespace OpenTelemetry.Instrumentation;
+
+public static class SqlProcessor
```

Comment:
nit: Don't absolutely love the name. SqlTextSanitizer?

Reply:
I know... I'll continue to consider the name. I originally had `SqlSanitizer` but in an upcoming PR it's going to do more than sanitization. It will also parse the operation and collection from the text to generate the db.query.summary.